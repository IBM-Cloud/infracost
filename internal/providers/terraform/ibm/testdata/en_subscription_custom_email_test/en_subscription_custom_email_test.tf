terraform {
  required_providers {
    ibm = {
      source = "IBM-Cloud/ibm"
    }
  }
}

locals {
  event_notifications = {
    plans = ["lite", "standard"],
  }
}

resource "ibm_resource_instance" "event_notifications" {
  for_each          = toset(local.event_notifications.plans)
  name              = "event-notifications-${each.value}"
  location          = "us-south"
  plan              = each.value
  resource_group_id = "default"
  service           = "event-notifications"
}

resource "ibm_en_subscription_custom_email" "subscription_custom_email_lite" {
  instance_guid  = ibm_resource_instance.event_notifications["lite"].guid
  name           = "Custom Domain E-mail Subscription"
  description    = "Custom Domain E-mail Subscription for Event Notifications"
  destination_id = "somedestinationid"
  topic_id       = "sometopicid"
}

resource "ibm_en_subscription_custom_email" "subscription_custom_email_standard" {
  instance_guid  = ibm_resource_instance.event_notifications["standard"].guid
  name           = "Custom Domain E-mail Subscription"
  description    = "Custom Domain E-mail Subscription for Event Notifications"
  destination_id = "somedestinationid"
  topic_id       = "sometopicid"
}

resource "ibm_en_subscription_custom_email" "subscription_custom_email_standard_no_usage" {
  instance_guid  = ibm_resource_instance.event_notifications["standard"].guid
  name           = "Custom Domain E-mail Subscription"
  description    = "Custom Domain E-mail Subscription for Event Notifications"
  destination_id = "somedestinationid"
  topic_id       = "sometopicid"
}
