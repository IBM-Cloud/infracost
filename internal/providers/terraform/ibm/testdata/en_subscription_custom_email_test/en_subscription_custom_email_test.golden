
 Name                                                                                Monthly Qty  Unit                    Monthly Cost 
                                                                                                                                       
 ibm_en_subscription_custom_email.subscription_custom_email_lite                                                                       
 ├─ Outbound Custom Domain E-mail Messages (Lite plan) (Max. 20)                              20  Messages                       $0.00 
 └─ Outbound Transmitted E-mail Messages (Lite plan)                                           1  GB                             $0.00 
                                                                                                                                       
 ibm_en_subscription_custom_email.subscription_custom_email_standard                                                                   
 ├─ Outbound Custom Domain E-mail Messages (Standard plan)                                 1,000  Messages                       $0.15 
 └─ Outbound Transmitted E-mail Messages (Standard plan)                                       1  GB                             $0.12 
                                                                                                                                       
 ibm_en_subscription_custom_email.subscription_custom_email_standard_no_usage                                                          
 ├─ Outbound Custom Domain E-mail Messages (Standard plan)                     Monthly cost depends on usage: $0.00015 per Messages    
 └─ Outbound Transmitted E-mail Messages (Standard plan)                       Monthly cost depends on usage: $0.12 per GB             
                                                                                                                                       
 ibm_resource_instance.event_notifications["lite"]                                                                                     
 └─ Lite Plan                                                                                  1  Instance                       $0.00 
                                                                                                                                       
 ibm_resource_instance.event_notifications["standard"]                                                                                 
 └─ Ingested Events                                                            Monthly cost depends on usage: $1.08 per Million Events 
                                                                                                                                       
 OVERALL TOTAL                                                                                                                   $0.27 
──────────────────────────────────
5 cloud resources were detected:
∙ 5 were estimated, 2 of which include usage-based costs, see https://infracost.io/usage-file